File: 30-unexplained-costs/python-sdk/unexplained.py
import os, requests
API = "https://api.costkatana.com/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def get_unexplained_costs():
    res = requests.get(f"{API}/analytics/unexplained-costs",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"💰 Unexplained costs: {res.json()['data']['total']}")
    return res.json()["data"]

def get_cost_anomalies():
    res = requests.get(f"{API}/analytics/cost-anomalies",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"⚠️ Anomalies found: {len(res.json()['data'])}")
    return res.json()["data"]

File: 30-unexplained-costs/npm-package/unexplained.ts
import axios from 'axios';
const API = 'https://api.costkatana.com/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function getUnexplainedCosts() {
  const res = await axios.get(`${API}/analytics/unexplained-costs`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('💰 Unexplained costs:', res.data.data.total);
  return res.data.data;
}

export async function getCostAnomalies() {
  const res = await axios.get(`${API}/analytics/cost-anomalies`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('⚠️ Anomalies found:', res.data.data.length);
  return res.data.data;
}
